lib/workers/repository/model/commit-message.ts

```diff
@@ -21,34 +31,54 @@ export class CommitMessage {
     return `${prefix}${CommitMessage.SEPARATOR}`;
   }
 
-  public setMessage(message: string): void {
-    this.message = (message || '').trim();
+  toString(): string {
+    const parts: ReadonlyArray<string | undefined> = [
+      this.title,
+      this.body,
+      this.footer,
+    ];
+
+    return parts.filter(Boolean).join('\n\n');
   }
 
-  public setCustomPrefix(prefix?: string): void {
-    this.prefix = (prefix ?? '').trim();
+  get title(): string {
+    return [CommitMessage.formatPrefix(this.prefix), this.formatSubject()]
+      .join(' ')
+      .trim();
   }
 
-  public setSemanticPrefix(type?: string, scope?: string): void {
-    this.prefix = (type ?? '').trim();
+  toJSON(): CommitMessageJSON {
```

Comment:
Is this only for tests? Then we should find another way. We shouldn't have a function which is only called on tests.

Reply:
For now yes, but it will be used in the next PR. This functionality introduced instead of giving direct access to commit message fields or implementing getter per each field. Open for good suggestions

Reply:
Where will it be used later? I'm still confused why we need that.

I know `JSON.stringify` will call that, but then it can return `unknown` because it's not really public API.

Reply:
I'll prefer to make the properties public with getter and setter and remove this method.

eg:
```ts
get body(): string {
  return this._body;
}

set body(body: string) {
  this._body = body.trim(); // do we need null and undefined?
}

Reply:
I would say it is quite subjective. I don't see any problem to have this `toJSON` method.

Reply:
Sid effects: this method would be called by `JSON.stringify` if an instance is serialized at any time.

I'm still not happy about it. Also private typescript fields are not really private, that's only a compiler check.

Reply:
It is not side effect, it is expected behaviour. If one day we need to serialize some data structure contains instance of `CommitMessage`, it will be serialized properly.

Reply:
> Also private typescript fields are not really private, that's only a compiler check.

If we can migrate from TypeScript's private modifier (`private`) to ECMAScript's private modifier[^1] (`#`) then things can be private?

[^1]: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-3.html#ecmascript-private-class-elements

Reply:
Refactored to private JS fields